sql/src/main/java/io/crate/user/UserActions.java

```diff
@@ -40,10 +40,14 @@ private UserActions() {
 
     }
 
-    public static SecureHash generateSecureHash(CreateUserAnalyzedStatement statement, Row parameters) throws GeneralSecurityException {
+    public static SecureHash generateSecureHash(CreateUserAnalyzedStatement statement, Row parameters) throws GeneralSecurityException, IllegalArgumentException {
         SecureString pw = getUserPasswordProperty(statement.properties(), parameters);
         SecureHash secureHash = null;
         if (pw != null) {
+            if (pw.length() == 0) {
+                pw.close();
+                throw new IllegalArgumentException("Password must not be empty");
```

Comment:
Would use 

```java
try (SecureString pw = getUserPasswordProperty(statement.properties(), parameters)) {
....
}
```

This will automatically close pw in any case.